scripts/bert/finetune_mnli.py

```diff
@@ -0,0 +1,244 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint:disable=redefined-outer-name,logging-format-interpolation
+
+import time
+import argparse
+import random
+import logging
+import warnings
+import numpy as np
+import mxnet as mx
+from mxnet import gluon
+import gluonnlp as nlp
+from gluonnlp.model import bert_12_768_12
+from gluonnlp.data import BERTTokenizer
+
+from bert import BERTClassifier
+from dataset import MNLIDataset, BERTDatasetTransform
+
+parser = argparse.ArgumentParser(description='BERT sentence pair classification example.'
+                                             'We fine-tune the BERT model on MRPC')
+parser.add_argument('--epochs', type=int, default=3, help='number of epochs, default is 3')
+parser.add_argument('--batch_size', type=int, default=32,
+                    help='Batch size. Number of examples per gpu in a minibatch, default is 32')
+parser.add_argument('--dev_batch_size', type=int, default=8,
+                    help='Batch size for dev set, default is 8')
+parser.add_argument('--optimizer', type=str, default='bertadam',
+                    help='Optimization algorithm, default is bertadam')
+parser.add_argument('--lr', type=float, default=5e-5,
+                    help='Initial learning rate, default is 5e-5')
+parser.add_argument('--warmup_ratio', type=float, default=0.1,
+                    help='ratio of warmup steps used in NOAM\'s stepsize schedule, default is 0.1')
+parser.add_argument('--log_interval', type=int, default=100, help='report interval, default is 100')
+parser.add_argument('--max_len', type=int, default=80,
+                    help='Maximum length of the sentence pairs, default is 80')
+parser.add_argument('--seed', type=int, default=2, help='Random seed, default is 2')
+parser.add_argument('--accumulate', type=int, default=None, help='The number of batches for '
+                    'gradients accumulation to simulate large batch size. Default is None')
+parser.add_argument('--gpu', action='store_true', help='whether to use gpu for finetuning')
+args = parser.parse_args()
+
+logging.getLogger().setLevel(logging.DEBUG)
+logging.info(args)
+
+batch_size = args.batch_size
+dev_batch_size = args.dev_batch_size
+lr = args.lr
+accumulate = args.accumulate
+log_interval = args.log_interval * accumulate if accumulate else args.log_interval
+if accumulate:
+    logging.info('Using gradient accumulation. Effective batch size = %d', accumulate * batch_size)
+
+# random seed
+np.random.seed(args.seed)
+random.seed(args.seed)
+mx.random.seed(args.seed)
+
+ctx = mx.cpu() if not args.gpu else mx.gpu()
+
+# model and loss
+dataset = 'book_corpus_wiki_en_uncased'
+bert, vocabulary = bert_12_768_12(dataset_name=dataset,
+                                  pretrained=True, ctx=ctx, use_pooler=True,
+                                  use_decoder=False, use_classifier=False)
```

Comment:
Can we also add an option to load from a specific checkpoint file? For example, users can specify `--load_checkpoint ~/.mxnet/bert/xxx.params`

Reply:
@eric-haibin-lin 
GLUE task only requires 3 or 5 epochs for fine-tuning, I don't think `checkpoint` requires currently.

Reply:
Good question. What I mean is that if someone uses our pre-training script to further pre-train on their dataset (work in progress, not merged yet), they're interested in loading the bert checkpoint and finetune on other tasks

Reply:
Yes, adding this option will make the script work for a more specific situation, and this is easy to implement. I can add it later.